src/aapns/connection.py

```diff
@@ -0,0 +1,434 @@
+import json
+import ssl
+from asyncio import (
+    CancelledError,
+    Event,
+    Future,
+    TimeoutError,
+    create_task,
+    open_connection,
+    wait_for,
+)
+from contextlib import suppress
+from dataclasses import dataclass, field
+from logging import getLogger
+from math import inf
+from time import time
+from typing import Dict, List, Optional
+from urllib.parse import urlparse
+
+import h2.config
+import h2.connection
+import h2.settings
+
+from .errors import Blocked, Closed, FormatError, ResponseTooLarge, Timeout
+
+# Apple limits APN payload (data) to 4KB or 5KB, depending.
+# Request header is not subject to flow control in HTTP/2
+# Data is subject to framing and padding, but those are minor.
+MAX_NOTIFICATION_PAYLOAD_SIZE = 5120
+REQUIRED_FREE_SPACE = 6000
+# OK response is empty
+# Error response is short json, ~30 bytes in size
+MAX_RESPONSE_SIZE = 2 ** 16
+# Inbound connection flow control window
+# It's quite arbitrary, guided by:
+# * concurrent requests limit, server limit being 1000 today
+# * expected response size, see above
+CONNECTION_WINDOW_SIZE = 2 ** 24
+logger = getLogger(__package__)
+
+
+@dataclass
+class Channel:
+    sid: int
+    fut: Optional[Future]
+    ev: List[h2.events.Event] = field(default_factory=list)
```

Comment:
```suggestion
    stream_id: int
    events_ready: Optional[Future]
    h2_event_queue: List[h2.events.Event] = field(default_factory=list)
```

(maybe the last should be a deque?)

Reply:
There was a bug there, thanks!

Here's how it works now, writer:
```py
for event in ...:
    sid = ...
    self.channels[sid].events.append(event)
```

And reader:
```py
for event in channel.events:
    ...
del channel.events[:]
```